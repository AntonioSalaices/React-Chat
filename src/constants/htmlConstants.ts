import { ChangeEvent, ComponentProps } from 'react';
import { Card } from '@Components/Core';

export enum HTMLElements {
  Input = 'input',
  Span = 'span',
}

export enum HTMLType {
  text = 'text',
  number = 'number',
  date = 'date',
  color = 'color',
  range = 'range',
  password = 'password',
}

export type InputType = typeof HTMLType;

export type SingleInputType = {
  [K in keyof InputType]: {};
}[keyof InputType];

export type InputProps = ComponentProps<HTMLElements.Input>;
export type SpanProps = ComponentProps<HTMLElements.Span>;

export type CardProps = ComponentProps<typeof Card>;

export type OnChangeType = ChangeEvent<HTMLInputElement>;

export type InputRef = HTMLInputElement;
