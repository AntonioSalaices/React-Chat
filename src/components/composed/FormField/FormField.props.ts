import { DebouncedFunc } from 'lodash';
import I18n from 'i18n-js';
import { TxKeyPath } from '@Translate/i18n';
import { OnChangeType, SingleInputType } from '@Constans/htmlConstants';

export interface FormFieldProps {
  /**
   * Current selected value depicted by pagination attribute
   */
  search?: string;
  /**
   * Handle fuction to listen every input change with delay
   */
  handleChange: DebouncedFunc<({ target: { value } }: OnChangeType) => void>;
  /**
   * Optional options to pass to i18n, Useful for interpolation
   * as well as explicitly setting locale or translation
   */
  txOptions?: I18n.TranslateOptions;
  /**
   * The text to display if not using `tx` or nested components
   */
  text?: string;
  /**
   * Text which is looked up via i18n
   */
  tx?: TxKeyPath;
  /**
   * sub text which is looked up via i18n
   */
  subTx?: TxKeyPath;
  /**
   * The sub text to display if not using `tx` or nested components
   */
  subText?: string;
  /**
   * Icon to display inside of the input
   */
  icon?: JSX.Element;
  /**
   * Type of the input
   */
  type: SingleInputType;
}
