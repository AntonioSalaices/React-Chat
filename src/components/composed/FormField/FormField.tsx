import Input from '@Components/basics/Input';
import { FormFieldProps } from './FormField.props';
import { translate } from '@Translate/translate';

const FormField: React.FC<FormFieldProps> = ({
  type,
  icon,
  search,
  handleChange,
  txOptions,
  text,
  tx,
  subTx,
  subText,
}) => {
  const i18Text = tx && translate(tx, txOptions);
  const content = i18Text || text;
  const i18SubText = subTx && translate(subTx, txOptions);
  const subContent = i18SubText || subText;

  return (
    <div className="col-11-xs col-11-sm col-5-md col-5-xl">
      <Input
        icon={icon}
        testID="searchInput"
        value={search}
        type={type}
        placeholder={content}
        subText={subContent}
        onChange={handleChange}
      />
    </div>
  );
};

export default FormField;
