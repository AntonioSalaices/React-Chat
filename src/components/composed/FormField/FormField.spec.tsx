import { render, RenderResult } from '@testing-library/react';
import FormField from './FormField';
import { debounce } from 'lodash';
import { HTMLType } from '@Constans/htmlConstants';

describe('<FormField />', () => {
  let component: RenderResult;
  const onChange = debounce(jest.fn(), 300);
  beforeEach(() => {
    component = render(<FormField type={HTMLType.text} icon={<i></i>} text="test" handleChange={onChange} />);
  });

  test('should render correctly', () => {
    const { container } = component;

    expect(container).toMatchSnapshot();
  });
});
