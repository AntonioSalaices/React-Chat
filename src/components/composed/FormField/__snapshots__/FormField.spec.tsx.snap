// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField /> should render correctly 1`] = `
<div>
  <div
    class="col-11-xs col-11-sm col-5-md col-5-xl"
  >
    <div
      class="gap-2"
    >
      <div
        class="input-icon"
      >
        <span>
          <i />
        </span>
      </div>
      <input
        class="input-primary"
        data-testid="searchInput"
        placeholder="test"
        type="text"
        value=""
      />
      <p
        class="text-primary-light-7"
      />
    </div>
  </div>
</div>
`;
